server/src/main/java/io/crate/planner/optimizer/costs/PlanStats.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.planner.optimizer.costs;
+
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+import io.crate.common.collections.Maps;
+import io.crate.expression.symbol.Literal;
+import io.crate.metadata.RelationName;
+import io.crate.planner.operators.Collect;
+import io.crate.planner.operators.CorrelatedJoin;
+import io.crate.planner.operators.Count;
+import io.crate.planner.operators.Get;
+import io.crate.planner.operators.GroupHashAggregate;
+import io.crate.planner.operators.HashAggregate;
+import io.crate.planner.operators.HashJoin;
+import io.crate.planner.operators.Insert;
+import io.crate.planner.operators.Limit;
+import io.crate.planner.operators.LogicalPlan;
+import io.crate.planner.operators.LogicalPlanVisitor;
+import io.crate.planner.operators.NestedLoopJoin;
+import io.crate.planner.operators.TableFunction;
+import io.crate.planner.operators.Union;
+import io.crate.planner.optimizer.iterative.GroupReference;
+import io.crate.planner.optimizer.iterative.Memo;
+import io.crate.planner.selectivity.SelectivityFunctions;
+import io.crate.sql.tree.JoinType;
+import io.crate.statistics.Stats;
+import io.crate.statistics.TableStats;
+import io.crate.types.DataTypes;
+
+public class PlanStats {
+    // used for testing
+    public static final PlanStats EMPTY = new PlanStats(new TableStats());
```

Comment:
Can we move this to the test cases? Looks like this should never be used in production code

Reply:
Yes, this makes sense.